lib/logstash/outputs/elasticsearch.rb

```diff
@@ -33,6 +33,24 @@
 # If using the default `protocol` setting ("node"), your firewalls might need
 # to permit port 9300 in *both* directions (from Logstash to Elasticsearch, and
 # Elasticsearch to Logstash)
+#
+# ## Retry Policy
+#
+# Not all events always make it successfully to Elasticsearch, and they are not always 
+# formatted correctly for the index they are targeting. So that we minimize loss of 
+# events, we have a specific retry policy in place.
+# 
+# Here are the rules of what is retried when:
+#
+# - Block and retry all events in bulk response that experiences transient network exceptions until
+#   a successful submission is received by Elasticsearch.
+# - retry subset of sent events which resulted in ES errors of a retryable nature which can be found 
+#   in RETRYABLE_CODES
+# - for events which returned retryable error codes, they will be pushed onto a separate queue for 
+#   retrying events. events in this queue will be retried a maximum of :max_retries times. The size of 
```

Comment:
Change maximum of :max_retries times => maximum of default 5 times (configurable through :max_retries times)